File: JAVA_CODE/sorting/BubbleSort.java
class BubbleSort
{
public static void sort(int arr[])
{
int e,f,m,gg;
m=arr.length-2;
while(m>=0)
{
e=0;
f=e+1;
while(e<=m)
{
if(arr[f]<arr[e])
{
gg=arr[e];
arr[e]=arr[f];
arr[f]=gg;
}
e++;
f++;
}
m--;
}
}
public static void main(String args[])
{
int arr[]={45,12,78,65,98,32,55,60,81,99};
sort(arr);
for(int i=0;i<arr.length;i++) System.out.print(arr[i]+" ");
}
}

File: generic/src/bubbleSort.c
#include<mg_sort.h>
#include<stdlib.h> //malloc,free
#include<string.h> //memcpy
void swap(void *a,void *b,int elementSize)
{
void *c=malloc(elementSize);
memcpy(c,a,elementSize);
memcpy(a,b,elementSize);
memcpy(b,c,elementSize);
free(c);
}
void bubbleSort(void *arr,int size,int sizeOfElement,int (*ptr2func)(void*,void*))
{
void *a,*b;
int m,e,f;
m=size-2;
while(m>=0)
{
e=0;
f=1;
while(e<=m)
{
a=arr+(f*sizeOfElement);
b=arr+(e*sizeOfElement);
if(ptr2func(a,b)<0)
{
swap(a,b,sizeOfElement);
}
e++;
f++;
}
m--;
}
}